plugin/testdata/custom-entity-check.lua

```diff
@@ -0,0 +1,24 @@
+local typedefs = require "kong.db.schema.typedefs"
+return {
+  name = "custom-entity-check",
+  fields = {
+    {protocols = typedefs.protocols},
+    {config = {
+        type = "record",
+        fields = {
+          {strategy = {type = "string", default = "localhost", one_of = { "localhost", "other"}}},
+          {other = {type = "record", fields = {{host = typedefs.host}}}}
+  }}}},
+  entity_checks = {{
+    custom_entity_check = {
+      field_sources = { "config" },
+      fn = function(entity)
+        local config = entity.config
+        print("config = " .. require("inspect")(config))
```

Comment:
leftover debug or intentional?

Reply:
maybe ;)

Reply:
rebased with fix to allow commit history to merge cleanly